src/test/librbd/crypto/test_mock_BlockCrypto.cc

```diff
@@ -35,12 +36,13 @@ struct TestMockBlockCrypto : public TestFixture {
       cryptor.block_size = cryptor_block_size;
       bc = new BlockCrypto<MockCryptoContext>(
               reinterpret_cast<CephContext*>(m_ioctx.cct()), &cryptor,
-              block_size);
+              block_size, data_offset);
       expectation_set = new ExpectationSet();
     }
 
     void TearDown() override {
       delete expectation_set;
+      bc->put();
```

Comment:
Nit: if you wrap the variable declaration in `ceph::ref` you can avoid this `put`